docs/kb/rules/using-semgrep-rule-schema-in-vscode.md

```diff
@@ -0,0 +1,78 @@
+
+### Associate the Semgrep rule schema
+
+Once the extension is installed, associate the Semgrep rule schema with Semgrep YAML rule definitions you are working on in VS Code using one of two recommended methods:
+
+1. Directly in the YAML file
+2. Using `yaml.schemas` in your VS Code `settings.json` file
+
+We recommend taking a look at the [extension overview section on associating schemas](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml#associating-schemas) as a precursor to gain a preliminary understanding before proceeding.
+
+#### Associating a schema directly in the YAML file
+
+To associate the schema easily and directly with a Semgrep YAML rule file, include the following line at the top of the file:
+
+    # yaml-language-server: $schema=https://json.schemastore.org/semgrep.json
+
+The drawback to this method is that it must be done independently for each YAML rule file.
+
+#### Associating a schema to a glob pattern via `yaml.schemas`
+
+Again, we highly recommend first reading the [extension overview](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml#associating-a-schema-to-a-glob-pattern-via-yaml.schemas) as a supplement to this article to better understand how YAML schemas are handled by the extension.
+
+To associate the Semgrep rule schema via `yaml.schemas` in your VS Code `settings.json` file (on MacOS), go to:
+
+    Code -> Settings -> Settings -> Extensions -> YAML
+
+In the YAML extension settings, scroll down to `Yaml: Schemas` and click `Edit in settings.json`, as shown below:
+
+![MacOS VS Code YAML extension settings](/img/kb/vscode-yaml-schemas.png)
+
+This opens the `settings.json` file with an empty `yaml.schemas` object already entered, ready to be defined. For example, consider the following `yaml.schemas` definition:
+
+```json
+"yaml.schemas": {
+    "https://json.schemastore.org/semgrep.json": "Downloads/semgrep_rules/*.yaml"
+}
+```
+
+This associates the schema defined on the left side of the colon (`:`) with files matching the glob pattern on the right. The glob pattern matches any `.yaml` file located in a directory structure that matches `Downloads/semgrep_rules/`. The desired glob pattern differs for varying operating systems and should reflect where you are storing Semgrep YAML rule files.
+
+After completing the configuration for `yaml.schemas`, open a Semgrep rule YAML file to verify that a notice shows at the top similar to this one:
+
+![Example Semgrep YAML rule file with schema defined](/img/kb/vscode-yaml-schema-example-file.png)
+
+This indicates that you've successfully associated the Semgrep rule schema with your Semgrep rule YAML file(s).
```

Comment:
```suggestion
This indicates that you've successfully associated the Semgrep rule schema with your Semgrep rule YAML files.
```

Reply:
I think (s) makes sense, as the user may only care about 1 file, or multiple.